primary/src/tests/core_tests.rs

```diff
@@ -193,13 +226,31 @@ async fn process_header_missing_payload() {
         /* rx_proposer */ rx_headers,
         tx_consensus,
         /* tx_proposer */ tx_parents,
+        metrics.clone(),
     );
 
-    // Send a header to the core.
-    let header = Header {
-        payload: [(BatchDigest::default(), 0)].iter().cloned().collect(),
-        ..header()
-    };
```

Comment:
That was a bug on the original test - the signature of the header is invalid. The test was passing, but for the wrong reasons (the header was never processed not because the payload was missing but because was invalid - so a silent error was arisen).